Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -94,9 +102,14 @@ ISymbolTableRelocator symbolTableRelocator
       if (!documents.TryGetValue(documentUri, out var databaseEntry)) {
         throw new ArgumentException($"the document {documentUri} was not loaded before");
       }
-      if (documentChange.TextDocument.Version != databaseEntry.Version + 1) {
-        throw new InvalidOperationException($"the updates of document {documentUri} are out-of-order");
+
+      var oldVer = databaseEntry.Version;
+      var newVer = documentChange.TextDocument.Version;
+      if (oldVer >= newVer) {
```

Comment:
Might be nice to add a link to LSP documentation about how the version should be handled. https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/specification-3-16.md?plain=1#L1195

Reply:
Good idea. Done.